### DISCUSSION TITLE
Why TTL for queue's events is Infinity. How to add TTL ?

### DISCUSSION THREAD

**Discussion by ruhankhandakar** [taskforcesh/bullmq#2442] [Q&A]
<img width="967" alt="Screenshot 2024-02-22 at 12 52 28 PM" src="https://github.com/taskforcesh/bullmq/assets/52363470/11f38529-6e74-4aa6-90d1-a890f2a23101">
on the uploaded screenshot we can see the `TTL` of `PROSPEO_QUEUE` event is `Forever`. I think in this way soon all my redis memory will be used by the `finished` events. How to add `TTL` or `remove` it from redis.

**Comment by ruhankhandakar** [comment]
Any reply will be appreciated, please.

**Reply by manast** [reply]
TTL is a Redis key option. It is forever since that key will not expire.

**Reply by manast** [reply]
If you want to reduce the number of events thats kept in that key you can change the default 10000 to any other value when you instantiate the queue: https://api.docs.bullmq.io/interfaces/v5.QueueOptions.html#streams
```ts
const myQueue = new Queue('myqueue', { streams: { events: { maxLen: 100 } } });
```

**Reply by ruhankhandakar** [reply]
<img width="809" alt="image" src="https://github.com/taskforcesh/bullmq/assets/52363470/344c20e0-f35b-4549-a522-63076ad9ec1c">
I am using NestJS, where to set the TTL limit ? so that the events get removed automatically from my Redis.

**Reply by manast** [reply]
There is no TTL in BullMQ, please read my previous answer.
